### ISSUE TITLE
Games launched with proton do not detect controllers

### ISSUE THREAD

**Issue by Tuerai** [ValveSoftware/Proton#1549]
# Compatibility Report
- Name of the game with compatibility issues:
Burnout Paradise
Dragon Quest XI
- Steam AppID of the game:
24740
742120

## System Information
- GPU: 1050ti
- Driver/LLVM version: nvidia 396.54-6
- Kernel version: 4.18.9.arch1-1
- Link to full system information report as [Gist](https://gist.github.com/):
https://gist.github.com/Tuerai/1ea2f0b4552e4d33fc8406c29cf4f599
- Proton version: 3.7-6

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-742120.log](https://github.com/ValveSoftware/Proton/files/2408242/steam-742120.log)

## Symptoms 
Tested with an Xbox One S controller, and a PS Dual Shock 4 controller, both connected wired. They work in native steam games. They work in games where I use Lutris with WINE. They work in Steam Big Picture Mode. They are not detected at all in games I launch through proton. Have tested in Burnout Paradise and Dragon Quest XI.

## Reproduction

I have steam installed on Arch Linux. I have launched it as both "Steam Native" and as "Steam Runtime" with the same behavior on both (works in BPM and not detected by Proton-launched game).

I tried both with and without xboxdrv/ds4drv installed, and had the same behavior as well.

**Comment by unomar** [comment]
Same issues here.  Tried both a Steam Controller and wired Xbox 360 gamepad.  Running Fedora 28 with the latest beta steam client.  No controller inputs detected in Burnout Paradise.  Both controllers work fine in big picture mode and native Linux games.

**Comment by dearlygparted** [comment]
> I have steam installed on Arch Linux. I have launched it as both "Steam Native" and as "Steam Runtime" with the same behavior on both (works in BPM and not detected by Proton-launched game).

Similar issue here. Xbox 360 wired controller works in Steam BPM and in native Steam for Linux games, but not in Final Fantasy VII, however it works in Kingdom Come Deliverance.

**Comment by grahammitchell** [comment]
Same issue here. Trying to play _Tales from the Borderlands_ through Proton 3.7-6 using my Steam Controller and the game only shows options for keyboard and mouse. It doesn't matter what config I load in.

It's a shame, because the game works great through Proton otherwise; it would be really nice to sit on the couch and play through it finally.

**Comment by junglerobba** [comment]
The latest [Steam Client Beta Update](https://steamcommunity.com/groups/SteamClientBeta#announcements/detail/1691556695797786203) seems to fix this. I had the same issue with Monster Hunter World, both with Steam and Xbox One Controller, and those work fine now.

**Comment by grahammitchell** [comment]
I _just_ updated to the latest Steam Client Beta, and this still isn't fixed for me. _Tales from the Borderlands_ still doesn't detect that a controller is plugged in.

**Comment by jonathansty** [comment]
Experience this issue when launching monster hunter world although this issue doesn't seem to be happening with Castlevania: Lords of shadows.

**Comment by ghost** [comment]
Right click on the game in your library and make sure your mapping is configured correctly for your controller. Also check "Controller Settings" in Steam.

**Comment by grahammitchell** [comment]
> make sure your mapping is configured correctly for your controller

What?

**Comment by ghost** [comment]
I'm having the same problem here on multiple games(Monster Hunter World, Darksiders Warmaster edition and Nier Automata)

**Comment by Rabcor** [comment]
Hmm my logitech rumblepad set to xinput mode works fine.

**Comment by bradleypariah** [comment]
Having the same problem with an official wireless 360 controller.

Controller still works perfectly in SteamOS games, it works in Big Picture, but it doesn't work in Proton games anymore.

When I first installed Fallout: New Vegas, I had opted for Proton v3.14 (I think?) in Steam Play settings, but the controller didn't work, so I fell back to an older version (3.7-8, I believe), and the controller started working.  It worked for The Witcher 3 and Fallout 3, as well.  A Steam update came out, and now it doesn't matter if I'm in the Steam beta update or not.  It doesn't matter which version of Proton I choose.  None of the controller configurations/templates work, except for one  - if I choose the template that emulates a keyboard and mouse.

Kubuntu 18.04 64-bit, i7-2600S octa-core CPU, 8GB DDR3 RAM, Nvidia gtx680

**Comment by bradleypariah** [comment]
Interesting revelation.  Found a fix.

Step One: Open Big Picture mode, then navigate to Settings > Controller Settings.  Check Xbox Configuration Support

Step Two: (Still in BPM) Navigate to Fallout: New Vegas > Manage Game > Controller Options > Steam Input Per-Game Setting > Set to: "Forced Off"

Profit.  Still running Proton 3.7-8.  I'll check back on the topic if the fix is only temporary.

**Comment by 200Puls** [comment]
@bradleypariah 
I just had the same problem. 

My setup is a bit different. I am using a PS4 controller, Ubuntu 18.10, Proton 3.16.6 beta and NBA 2K Playgrounds 2 is the game in which the controller was not recognized.

Your suggested workaround fixed it also for me. So it seems to be a Steam problem.

**Comment by grahammitchell** [comment]
I just tried bradleypariah's workaround and it didn't make any difference. Controller still not recognized for me in _Tales from the Borderlands_.

Using an Xbox One controller wirelessly using [xpadneo](https://github.com/atar-axis/xpadneo) and Proton 3.7-8 (non-beta).

These Proton games work out-of-the-box with the controller:

* [Alan Wake](https://store.steampowered.com/app/108710/Alan_Wake/)
* [DOOM](https://store.steampowered.com/app/379720/DOOM/)
* [Gauntlet™ Slayer Edition](https://store.steampowered.com/app/258970/Gauntlet_Slayer_Edition/)
* [The Witness](https://store.steampowered.com/app/210970/The_Witness/)

The Xbox controller also works great with Linux native titles (not using Proton):

* [FEZ](https://store.steampowered.com/app/224760/FEZ/)
* [Mark of the Ninja Remastered](https://store.steampowered.com/app/860950/Mark_of_the_Ninja_Remastered/)
* [Rise of the Tomb Raider](https://store.steampowered.com/app/391220/Rise_of_the_Tomb_Raider/)

The controller is _not_ detected in the following game under Steam Play/Proton:

* [Tales from the Borderlands](https://store.steampowered.com/app/330830/Tales_from_the_Borderlands/)

It won't detect the controller using bradleypariah's workaround or without. It also doesn't seem to matter whether I use the Xbox One controller or my Steam Controller.

**Comment by bradleypariah** [comment]
grahammitchell - Question, just to move the conversation forward - If you turn on XBox Configuration Support in BPM settings, and Force On Steam Input Per-Game Setting on the game, then download and apply a template, does that work?  What if you choose the keyboard and mouse template?  I'm thinking the devs might be curious to know which hand-off is getting dropped.

**Comment by grahammitchell** [comment]
Thanks for the suggestions.

> If you download and apply a template, does that work?

No. _Tales from the Borderlands_ still behaves like no controller is detected.

> What if you choose the keyboard and mouse template?

That _does_ work, in the sense that it allows me to move the mouse around using the right stick and click using right trigger. It's a fairly unpleasant gaming experience but I guess it _could_ be made to work if one were desperate to play it with a controller.

**Comment by MartinDueren** [comment]
I can confirm this. No proton launched game works with Steam controllers so far for me. Using the keyboard and mouse template works, but this is a tedious workaround. All native ones work flawlessly. The Steam overlay (and navigation in the overlay menus) works with the controller without problems.

**Comment by aroskuski** [comment]
I was having some issues like this with NieR: Automata a while ago.  I think I eventually worked around it by adding my user account to the input group, but it's been a while so I'm not 100% sure.

**Comment by Britaliope** [comment]
I can confirm that Steam Controller does not work (as a controller) for proton-launched games. Steam Controller works correctly on Big Picture and steam overlay menu but the game does not detetect the presence of a controller. 

Here is the log for Ori and the Blind Forest Definite Edition
[steam-387290.log](https://github.com/ValveSoftware/Proton/files/2734592/steam-387290.log)

**Comment by DanMan** [comment]
In Shadow of War it's also only detected, if you turn off the XBox Configuration Support in BPM settings. Proton 3.16-6B, Ubuntu 18.04.

Edit: in Dead Or Alive 6 it doesn't react to _any_ input once I'm past the start screen, which I am able to dismiss by pressing the enter key. Doesn't matter if Steam Input is enabled or not. Forcing it off makes the XB1 pad work though.

**Comment by just-Bri** [comment]
Has there been any update on this?
Want to make the switch to Linux only on my desktop, but I want to make sure I can use Proton for a couple games with my steam controllers. Specifically Death Squared, Lost Castle and Overcooked(1, haven't bought 2 yet).

**Comment by kparal** [comment]
Overcooked 1 works perfectly for me out of the box with both Steam Controller and Xbox One controller on Fedora 29.

**Comment by grahammitchell** [comment]
The vast majority of games work perfectly fine with controllers under Proton, which is why this bug hasn't been fixed yet. Only a very few games fail, and it's not clear why when so many others work.

In my comment from January 1, I list several other games that do work fine for me. Just not _Tales from the Borderlands_, in my case.

**Comment by neurodiverseEsoteric** [comment]
"Shooty Skies" doesn't work either, but it does in the native Linux version. Has anyone looked into this as being a directx issue rather than a proton issue? Why can't games just be rewritten in opengl instead of trying to get directx to work on linux?

**Comment by just-Bri** [comment]
Rewriting in OpenGL would be a full rewrite.
DXVK is a layer between directx and vulkan, so you don't need to rewrite.

**Comment by rmanosuthi** [comment]
I had this problem a while ago and fixed it by disabling kernel modules that handle controllers. Steam doesn't seem to like those modules.

I'm using Gentoo and I compiled my kernel manually, and games launched through Steam (i.e. Witcher 3) didn't detect my XB1S controller until I did that.

Not sure if this will solve anyone's problem though, I didn't test it with native Linux games.

**Comment by Azarilh** [comment]
> Interesting revelation. Found a fix.
> 
> Step One: Open Big Picture mode, then navigate to Settings > Controller Settings. Check Xbox Configuration Support
> 
> Step Two: (Still in BPM) Navigate to Fallout: New Vegas > Manage Game > Controller Options > Steam Input Per-Game Setting > Set to: "Forced Off"
> 
> Profit. Still running Proton 3.7-8. I'll check back on the topic if the fix is only temporary.

That worked for me on Monster Hunter World. Thank you!

**Comment by ekthomson** [comment]
Forced Off on the controller options also worked for Vessel.

**Comment by schmidtbag** [comment]
This is a pretty weird problem, because not only does this not affect every game, but it also seems to depend on your version of Proton.  For example, I tried playing the game Gas Guzzlers Extreme, and in Proton 4.2-3, none of the suggestions listed above work, whether I enable or disable them, in any combination.  But, if I downgrade to Proton 3.16, my controller works just fine.
I have a DS4 that I connect through Bluetooth, if that makes any difference.

**Comment by Jojonintendo** [comment]
Unfortunately I face a similar issue. My controller (Nintendo Switch Pro) works perfectly fine in native games, but Monster Hunter World won't work with it. I have tried to "Force off" and downgrade to 3.16-4, but it still doesn't work. With latest 4.2-5 the exact same happens. I have tried multiple combinations but the result is always the same, the controller won't work with Monster Hunter World.

Edit: my system is latest opensuse tumbleweed, kernel 5.1.4, steam 1556574584.

**Comment by fireedo** [comment]
e.g. Sine Mora EX, proton 4.2-5, Tumbleweed 5.1.4, Nvidia 430.14, Xbox wireless controller detected fine, launching game using option "forced off" success detected the controller but controller movement in game is very fast also in the menu selection. any idea?

**Comment by Jojonintendo** [comment]
After much trial and error, it seems that completely deactivating the Nintendo Switch Pro support actually makes it work somewhat in game. All buttons are recognized but the joysticks are wonky. If I turn the controller support back on, the blue LED turns on, but then it doesn't work at all in game. I have no idea what to try next.

Edit: I could finally solve it. The problem is that my user doesn't have access to the file /dev/uinput, until I load the uinput module. For more information see [here](https://github.com/rpmfusion/steam/pull/10).

**Comment by asgeir** [comment]
For me it looks like this was an issue with my CORSAIR Gaming K70 keyboard.

It appears that it has a joystick interface (I can't imagine why. For RGB control maybe?) so I had two /dev/input/js devices. /dev/input/js0 which would be my xbox controller and /dev/input/js1 which would be my keyboard.

I created a udev rule to block the joystick interface in my keyboard and suddenly controller input is working again and I'm back to a single /dev/input/js device.

```
SUBSYSTEM=="usb", ACTION=="add", ATTRS{idVendor}=="1b1c", ATTRS{idProduct}=="1b09", ATTR{bInterfaceClass}=="03", ATTR{bInterfaceSubClass}=="01", ATTR{bInterfaceProtocol}=="01", ATTR{authorized}="0"
```

**Comment by AaronLMorgan** [comment]
> After much trial and error, it seems that completely deactivating the Nintendo Switch Pro support actually makes it work somewhat in game. All buttons are recognized but the joysticks are wonky. If I turn the controller support back on, the blue LED turns on, but then it doesn't work at all in game. I have no idea what to try next.
> 
> Edit: I could finally solve it. The problem is that my user doesn't have access to the file /dev/uinput, until I load the uinput module. For more information see [here](https://github.com/rpmfusion/steam/pull/10).

I wanted to second this solution for any other OpenSuse Tumbleweed users.

Tumbleweed Version: 20190612 (Latest), Kernel: 5.1.7-1, Steam Version: 1560727279 (Latest Beta Build)

My Steam Controller was only working on my desktop and while using Big Picture Mode. It was not working while playing Doom or Sekiro through Proton 4.2-7. After loading the uinput kernel module using modprobe, the controller began to work in both of these games. Thanks for the solution, Jojonintendo!

**Comment by WoutLin** [comment]
I have had issues with game controllers working on Steam Proton since it became available.  I had been using Steam on PlayOnLinux prior, but when Proton became available I wished to use it solely.  However I could not get my controllers to work for Burnout Paradise: The Ultimate Box and Flatout Ultimate Carnage.  Having fixed controller issues using Steam on PlayOnLinux I didn't need to spend time with Proton.  The controllers I use:  Saitek P2500 and Xiaoji Gamesir-G3w.  Both dual stick gamepads.  
Using PlayOnLinux I used wine controller config app to select either xinput or dinput and disable the other and that made the controllers work.  The controllers worked almost perfect, fully configurable and identified by the games.
Recently I updated Nvidia video drivers and had a booting issue.  So after no quick fix I updated to Linux Mint 19.1 (from Linux Mint 18.3).  I did a clean install from usb stick.  Long story short, once Steam was installed, using the Xiaoji Gamesir-G3w gamepad, without doing any controller calibration, mapping or anything else, it was identified and worked perfectly by both games.  Just had to set up the in-game controller settings.  Also worked on DiRT Rally, but not as perfect, buttons being incorrect, but totally useable.  Great.  
Next day I fire up Steam.  Flatout does not detect the gamepad.  The buttons work to navigate the menus but game controller settings are totally missing.  Burnout the gamepad name is there with previous settings but they won't respond to input.
DiRT Rally I can get to work.  Previously I mapped the gamepad with SDL2 tool and saved the result to a text file. DiRT would not recognize my rt joystick.  I used jstest and evtest and both provide output when actions are performed on the gamepad.  Proving linux is getting inputs.  I found .../common/DiRT Rally/share/inputdevices.json was a library of input devices.  I copied  the PS4 gamepad section and replaced  :     {
        "Name": "Xiaoji Gamesir-G3w",
        "VendorID": "0x05ac",
        "ProductIDs": [
            "0x055b",
			"*"
        ],
        "Category": "Gamepad",
        "Type": "PS4Gamepad",
        "SDLMapping": "a:b2,b:b1,x:b3,y:b0,back:b8,guide:b12,start:b9,leftstick:b10,rightstick:b11,leftshoulder:b4,rightshoulder:b5,dpup:h0.1,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,leftx:a0,lefty:a1,rightx:a2,righty:a3,lefttrigger:b6,righttrigger:b7,"
    },
Now works perfect.
What ticks me off is that Flatout and Burnout worked perfect for 1 time and now they don't.  
I have tried many of the suggestions.  I have a keyboard that comes up as js0 or js1.  Tried suggestions to add rules to blacklist keyboard to be identified as js.  Tried unplugging keyboard and starting games.  Still didn't work.  Made rule to access to input, access logged in user.  Not work.
Any clues to why the gamepad was recognized only after being installed but not subsequently?

**Comment by WoutLin** [comment]
After spending many fruitless hours trying to get Burnout and Flatout to identify and map my game controller using Steam/Proton, my final method to get the games to work is install Steam via PlayOnLinux for each game. Then use wine control panel/game controller and disable "js for Burnout and "event" for Flatout.  I don't have to touch Steam controller settings and the games work fine.

**Comment by ghost** [comment]
> After spending many fruitless hours trying to get Burnout and Flatout to identify and map my game controller using Steam/Proton, my final method to get the games to work is install Steam via PlayOnLinux for each game. Then use wine control panel/game controller and disable "js for Burnout and "event" for Flatout. I don't have to touch Steam controller settings and the games work fine.

You can run `wine control` on the proton Wine prefix.

`WINEPREFIX="$HOME/.local/share/steamapps/compatdata/24740/pfx" wine control`

**Comment by djohnson1865** [comment]
> I had this problem a while ago and fixed it by disabling kernel modules that handle controllers. Steam doesn't seem to like those modules.
> 
> I'm using Gentoo and I compiled my kernel manually, and games launched through Steam (i.e. Witcher 3) didn't detect my XB1S controller until I did that.
> 
> Not sure if this will solve anyone's problem though, I didn't test it with native Linux games.

Wanted to see if I could get some more clarification on what modules exactly you disabled. Also running Gentoo and compiled my own kernel. I had controller support working with a 360 controller in The Witcher 3 for about a day before it started having issues again with a steam client update. Was previously running kernel 4.19.44, since I have upgraded to 5.1.12 with no change in behavior. Now with any proton <4.2.8. I can get the controller recognized and it works in BPM, but controls are mapped completely wrong in every game, Xbox button is start, D-pad doesn't work, and left click is select, etc... The mapping is consistent between all versions of proton except >4.2.8 and is consistent in The Witcher 3 and DOOM (the two games I have tried). Disabling the xpad module prevented Steam from detecting the controller at all. Also, when I did get the controllers working for a short time I was using I believe 4.2.8 and needed to use forced off with the steam controller settings for it to work. 

Update: About 10 minutes after posting this a steam client beta update triggered and fixed the controller issues I was having. Still requires forced off controller settings, but that seems to fix my problem for anyone else in the same situation.

**Comment by WoutLin** [comment]
> 


    After spending many fruitless hours trying to get Burnout and Flatout to identify and map my game controller using Steam/Proton, my final method to get the games to work is install Steam via PlayOnLinux for each game. Then use wine control panel/game controller and disable "js for Burnout and "event" for Flatout. I don't have to touch Steam controller settings and the games work fine.

You can run wine control on the proton Wine prefix.

WINEPREFIX="$HOME/.local/share/steamapps/compatdata/24740/pfx" wine control

>z0z0z, I tried your suggestion and was indeed able to change the game controller settings in the wine control panel.  However it has no effect on the game using my controller.  Still works with PlayOnLinux so it is not much of an issue anymore.

**Comment by WoutLin** [comment]
I just bought Wreckfest Appid 228380 and my controller is not identified.  Again I tried Steam on PlayOnLinux, but had to run 32 bit, and the controller is identified.  However this game is unplayable with bad slow graphics.  I just need to identify what is different between proton and wine that affects the game controller. Any help is appreciated.

**Comment by WoutLin** [comment]
Need For Speed Hot Pursuit Appid 47870 also not identifying my controller.  I opened wine control panel in both Hot Pursuit and Wreckfest proton directories and my game controller is identified and all inputs are registered in the test section.  For some reason when the games are run on Steam proton the controller is not identified.

**Comment by WoutLin** [comment]
Contacted Steam support.  Suggested in controller settings, select ONLY xbox configuration support (nothing else).  Now the controller is accepted as xbox controller and inputs can be mapped.  It works fine for Wreckfest Appid 228380, NFS Hot Pursuit Appid 47870 and Burnout Paradise Appid 24740.  However, Flatout Ultimate Carnage Appid 12360 has a standard xbox controller template that works but inputs cannot be customized.  Running from PlayOnLinux and using generic controller, my controller is identified and fully customizable.  

I had worked on this for a long time and never came across this suggestion before.  Is there official valve/steam documentation that I missed that addresses this?  

I call this not a solution but a work around since my controller is still not identified and is not 100% effective.

Thanks to Steam support for the suggestion.

**Comment by xinyazhang** [comment]
> Interesting revelation. Found a fix.
> 
> Step One: Open Big Picture mode, then navigate to Settings > Controller Settings. Check Xbox Configuration Support
> 
> Step Two: (Still in BPM) Navigate to Fallout: New Vegas > Manage Game > Controller Options > Steam Input Per-Game Setting > Set to: "Forced Off"
> 
> Profit. Still running Proton 3.7-8. I'll check back on the topic if the fix is only temporary.

I confirm this solves the Octopath Traveler (appid 921570) controller detection issue.

**Comment by ChrisHighwind** [comment]
> Interesting revelation. Found a fix.
> 
> Step One: Open Big Picture mode, then navigate to Settings > Controller Settings. Check Xbox Configuration Support
> 
> Step Two: (Still in BPM) Navigate to Fallout: New Vegas > Manage Game > Controller Options > Steam Input Per-Game Setting > Set to: "Forced Off"
> 
> Profit. Still running Proton 3.7-8. I'll check back on the topic if the fix is only temporary.

Can confirm this works on Sonic Adventure DX using the Dreamcast Conversion mod.

**Comment by tvjg** [comment]
Confirming that this worked for me with Spelunky too.

**Comment by DanMan** [comment]
Yes, it's weird. I have support for generic gamepads enabled in the basic controller settings, but games don't detect my no-name DInput pad _unless_ I go into the game's Steam properties and force off Steam Input (or I launch it through Big Picture mode). Then it responds to pad input and even still uses the correct button mapping as configured in Steam. The latter comes unexpected, since I forced it off in the properties. But at least it ends up the way I want it, so that's good.

Oh, and that's not just the case with Proton games, also native ones.

**Comment by JulianGro** [comment]
Running `wine control` in the prefix shows gamepad options.
Running `proton run control` in the prefix does not show gamepad options on proton 4.11-1
Could this be related to some controller issues?
Personally I cannot get my gamepad working in GTA V, but in No Man's Sky it just works out of the box.

**Comment by ghost** [comment]
@jug007 While having the controller plugged in, try right clicking the game in your Steam library and select "Edit controller configuration".

There is also global controller configuration configuration in the Steam setings.

Make sure you take a look at both of these. By the way you need the Steam Overlay for proper controller support.

**Comment by JulianGro** [comment]
Yeah it shows the controller and the right settings. The overlay also works.

**Comment by DanMan** [comment]
I've opened a ticket over here because it doesn't just affect Proton games for me: https://github.com/ValveSoftware/steam-for-linux/issues/6428

**Comment by ipkpjersi** [comment]
> Interesting revelation. Found a fix.
> 
> Step One: Open Big Picture mode, then navigate to Settings > Controller Settings. Check Xbox Configuration Support
> 
> Step Two: (Still in BPM) Navigate to Fallout: New Vegas > Manage Game > Controller Options > Steam Input Per-Game Setting > Set to: "Forced Off"
> 
> Profit. Still running Proton 3.7-8. I'll check back on the topic if the fix is only temporary.

This worked for me with both an Xbox 360 Wireless controller and Xbox One Bluetooth Wireless controller in NBA 2K Playgrounds 2.

**Comment by howdev** [comment]
Usually only the DS4 does not detect in some proton games. When use per game forced on this would make DS4 detect as Xbox controller. For Xbox controller should work by default because of the protocol.

**Comment by WoutLin** [comment]
I have been using my controller successfully for Wreckfest since following the advice of Steam Support: select ONLY xbox configuration support (nothing else). 

However after remapping controls to my liking and playing the game flawlessly, on shutdown and restart the controls are back to default.  Since I did not find any solutions to this issue after searching, I worked out a fix myself.  Maybe it can help someone else if I am not the only one that had this issue.

My fix for proton controller settings not saved after shutting off wreckfest: 

copy all files from: [GAME INSTALLATION FOLDER]/steamapps/compatdata/228380/pfx/drive_c/users/steamuser/My Documents/my games/wreckfest/76561197999597403/controllers

to: [STEAM INSTALLATION FOLDER]/userdata/[USER ID]/228380/local/wreckfest/controllers

**Comment by ultra-azu** [comment]
So I just wanted to say that I tried both the workaround of disabling Controller Configuration in the game, disabling it globally, disabling the xpad driver, but I still can't play Proton games with my Xbox One controller. Like everyone else, the controller works fine elsewhere. I'll provide any info needed.

**Comment by landsman** [comment]
Flatout not working for me too with configured Dual Shock 4 controller (usb / bluetooth too). It's pitty.

**Comment by WoutLin** [comment]
I recently got Flatout UC to work with my GameSir G3 gamepad.  I followed Steam support advice and in general controller settings, selected only xbox, everything else blank.  
Then in game controller settings I mapped the gamepad sticks and buttons to keyboard buttons used by the game.  Next in the game controller options I picked "keyboard".  If I select xbox controller the template of functions cannot be changed but using keyboard it is customizable.
I can use the gamepad fully customizable.  Left stick for steering.  Right stick throttle/brake.  Triggers for hand brake and nitrous.  Works perfect.

**Comment by landsman** [comment]
@WoutLin It's working with Xbox in settings, thanks!

**Comment by nenoro** [comment]
well here's another feedback, i use gentoo and it's a gambling with my x360 controller
Right now when i launch witcher 3 or volgarr both are my rats lab to test if my controller works, as i said right now controller doesn't work.

i tried to forced off, to turn on via steam big picture but nope still nothing, the module in the kernel is turned as module not as built-in (even i use built-in it doesn't work)

for a time i figure out steam linux runtime was the cause so i had to force to use proton 5.x.

However after i launch FM18 everything still work but like 2 days later i launch witcher 3 to try bam x360c isn't working but it is working for non proton games..

i tried to wake up from sleep the usb part but nope still nothing also i get rid of Proton by Glorius Eggroll and still nothing

**Comment by minneyar** [comment]
Just thought I'd chime it that I was having a similar issue with Bloodstained: Curse of the Moon 2.  I'm using an 8BitDo SN30 pro+ in Xinput mode, connected via USB.  It shows up at `/dev/input/js0`, is the only gamepad, and works fine in other games and in big picture mode, but the game did not recognize it at all.

I was able to fix it in a similar to way to other people here:
1. Go to the global Controller Settings in Steam and enable Xbox Configuration Support
2. Go to Steam's Controller Settings specifically for the game and set "Steam Input Per-Game Setting" to Forced Off

Now the game (using Proton 5.0-9) detects my gamepad and works fine.

**Comment by RazrFalcon** [comment]
@minneyar Thanks! Surprisingly, disabling "Steam Input Per-Game Setting" fixed a controller detection in Burnout Paradise.

**Comment by neurodiverseEsoteric** [comment]
Proton 5.13 seems to be better at detecting controllers, but for some reason does not detect the 'O' and 'R1' buttons on my  'Mad Cats FightStick PS3 Controller' even though the buttons work and wine control.exe and linux can detect them being pressed

**Comment by dj--alex** [comment]
Proton 5.13, Proton 5.11-GE, Proton 5.9-Ge (newest) + PS3 controller.
Not detected in ANY Steam-Proton games. On Big picture, or any other game (native, wine, steam-native) PS3 will works.
Shadow warrior 2, Hard reset, Layers of fear 2, Blades of time
These games works 2 months ago with PS3 Joysticks (i have 3 joysticks)  using Bluetooth.
I not update My Linux Mint 19.3 x64 to newest OS. 
Of course i try everything.  not only disabling Steam input Per game...i waste 20 days for this! 
In previous attempt to fix i delete and clear Steam, his cache, games, saves, ALL. and reinstall Steam only.
PS3 joysticks starts working,  2 months of working and now znowu not works. 
I change my ming and i don't publish my game on steam. Its useless without controller.
I cannot play with keyboard and mouse. Its not comfortable for me.

**Comment by howdev** [comment]
@dj--alex try Proton 5.0.10 works even hotplug or 4.x

**Comment by dj--alex** [comment]
> @dj--alex try Proton 5.0.10 works even hotplug or 4.x

no works. im already tested it. i retry test but Shadow warrior or other still no works.
Someone tells Buy Xbox joystick but im using PS3 and don't like xbox controller
I cannot play games on steam. unfortunately

**Comment by howdev** [comment]
@dj--alex Does it work in wired? I have dualshock4  and works with bluetooth and wired in Mint 19.3 and Mint 20. 
Make sure you have steam-devices package installed otherwise the controller won't be detected in Proton games.

**Comment by dj--alex** [comment]
> @dj--alex Does it work in wired? I have dualshock4 and works with bluetooth and wired in Mint 19.3 and Mint 20.
> Make sure you have steam-devices package installed otherwise the controller won't be detected in Proton games.

Its installed. Wired - No. I have installed RPCS3 may this can be problem? /etc/udev/rules.d/99-ds3-controllers.rules . Have a PS3 console too.
# DualShock 3 over USB
KERNEL=="hidraw", ATTRS{idVendor}=="054c", ATTRS{idProduct}=="0268", MODE="0666"

# DualShock 3 over Bluetooth
KERNEL=="hidraw*", KERNELS=="*054C:0268*", MODE="0666"

**Comment by howdev** [comment]
@dj--alex  if it doesn't even work with wired then, you really have problem. I don't have any emulator installed and I don't have any rules under udev/rules.d. So that could be your problem.

**Comment by dj--alex** [comment]
> @dj--alex if it doesn't even work with wired then, you really have problem. I don't have any emulator installed and I don't have any rules under udev/rules.d. So that could be your problem.

Really i found solution.   remove  /etc/udev/rules.d/99-ds3-controllers.rules  fix the problem, but it brokes completely RPCS3 and i cannot start God of war.  I must select which application steam or RPCS3 can use PS3 joystick .  It can be fixed for both applications can work normally with this udev rule?

**Comment by howdev** [comment]
@dj--alex If installing RPCS3 break Proton, then this is not Proton issue. Proton works by default with controllers.

**Comment by nenoro** [comment]
Proton 5.13 has fixed the issue let's hope 5.14 won't go crazy and kick the fix

**Comment by ghost** [comment]
Games: Skyrim SE
Controller: Xbox Elite 2 (wired)

Controller works fine in native application, is not detected when launching a Proton game

**Comment by rokam** [comment]
Issue confirmed on GTA5 with Proton7 and with SteamController

**Comment by andymason** [comment]
Just chiming in with my experience. A 8BitDo Pro 2 controller in x-input (Xbox) mode worked fine with most games, but Monster Hunter Rise refused to see any controller connected.

The solution for my situation was to re-enable Steam overlay, Settings > In-Game > Enable the Steam Overlay.

I had previously turned off the overlay because I thought I didn't need it. However, for some reason this prevented recognition of the controller, or maybe stopped the profile loading?

**Comment by rokam** [comment]
> Issue confirmed on GTA5 with Proton7 and with SteamController

I don't know what was wrong, but now it works again!

**Comment by octopoulpe** [comment]
Yeah. Steam Input was not working for me on every unreal engine game. I had previously disabled Steam Overlay globally.
Re-enabling it solves my problem. Thanks @andymason

**Comment by zdimension** [comment]
> Interesting revelation. Found a fix.
> 
> Step One: Open Big Picture mode, then navigate to Settings > Controller Settings. Check Xbox Configuration Support
> 
> Step Two: (Still in BPM) Navigate to Fallout: New Vegas > Manage Game > Controller Options > Steam Input Per-Game Setting > Set to: "Forced Off"
> 
> Profit. Still running Proton 3.7-8. I'll check back on the topic if the fix is only temporary.

This fixed Stray not detecting the controller on Debian 11 and Proton-GE 8.0-20240111c. Thanks!

**Comment by Rabcor** [comment]
Honestly this is just a default configuration issue I think. Sometimes you gotta mess with the controller options for the game in order to get controllers to work with it.

Also poeple should know if you use protontricks, your controller settings get ignored entirely, which may mean that a controller that steam doesn't detect but the os does would work, but generally it means controller support is shit if ur on protontricks compared to launching via steam. (tinkerlaunch uses the steam controller settings though, since it is launched through steam) 

For instance, for me to get a ps4 controller to work i had to reconfigure the game like this:

![image](https://github.com/ValveSoftware/Proton/assets/5684325/b3efec62-a1a4-49a0-b205-20836ba463a8)

By default, only  steam controller and remote play are enabled.

**Comment by RetroNomad** [comment]
Bumping this once more rather than opening a new issue. 

Happens with various titles and it seems both Proton official versions and GE have had this issue for a few months now personally. 
Hotfix, Experimental, 10.0-4 are the main pain points. 
Proton 9.0-4 and below tends to work. 

Device is recognized within Steam and can be fully tested and as mentioned works in lower official Proton versions.

**Comment by JimJafar** [comment]
> Proton 9.0-4 and below tends to work.

Oh my word, thank you! I have tried all the other solutions and dropping to 9.0-4 is the only thing that has worked for me. Monster Hunter Wilds is finally playable!
